MinPlatformPkg/Test/Library/TestPointCheckLib/StandaloneMmTestPointCheckLib.c

```diff
@@ -0,0 +1,482 @@
+  );
+
+GLOBAL_REMOVE_IF_UNREFERENCED EFI_MEMORY_DESCRIPTOR  *mUefiMemoryMap;
+GLOBAL_REMOVE_IF_UNREFERENCED UINTN                  mUefiMemoryMapSize;
+GLOBAL_REMOVE_IF_UNREFERENCED UINTN                  mUefiDescriptorSize;
+
+EFI_MEMORY_ATTRIBUTES_TABLE  *mUefiMemoryAttributesTable;
+
+GLOBAL_REMOVE_IF_UNREFERENCED ADAPTER_INFO_PLATFORM_TEST_POINT_STRUCT  mTestPointStruct = {
+  PLATFORM_TEST_POINT_VERSION,
+  PLATFORM_TEST_POINT_ROLE_PLATFORM_IBV,
+  { TEST_POINT_IMPLEMENTATION_ID_PLATFORM_SMM },
+  TEST_POINT_FEATURE_SIZE,
+  { 0 }, // FeaturesImplemented
+  { 0 }, // FeaturesVerified
+  0,
+};
+
+GLOBAL_REMOVE_IF_UNREFERENCED UINT8  mFeatureImplemented[TEST_POINT_FEATURE_SIZE];
+
+/**
+  This service verifies SMRR configuration at the End of DXE.
+
+  Test subject: SMRR.
+  Test overview: Verify SMRR is aligned and SMRR matches SMRAM_INFO.
+  Reporting mechanism: Set ADAPTER_INFO_PLATFORM_TEST_POINT_STRUCT.
+                       Dumps SMRR and SMRAM_INFO.
+
+  @retval EFI_SUCCESS         The test point check was performed successfully.
+  @retval EFI_UNSUPPORTED     The test point check is not supported on this platform.
+**/
+EFI_STATUS
+EFIAPI
+TestPointStandaloneMmEndOfDxeSmrrFunctional (
+  VOID
+  )
+{
+  EFI_STATUS  Status;
+  BOOLEAN     Result;
+
+  if ((mFeatureImplemented[TEST_POINT_INDEX_BYTE6_SMM] & TEST_POINT_BYTE6_SMM_END_OF_DXE_SMRR_FUNCTIONAL) == 0) {
+    return EFI_SUCCESS;
+  }
+
+  DEBUG ((DEBUG_INFO, "======== TestPointStandaloneMmEndOfDxeSmrrFunctional - Enter\n"));
+
+  Result = TRUE;
+  Status = TestPointCheckSmrr ();
+  if (EFI_ERROR (Status)) {
+    Result = FALSE;
+  }
+
+  if (Result) {
+    TestPointLibSetFeaturesVerified (
+      PLATFORM_TEST_POINT_ROLE_PLATFORM_IBV,
+      NULL,
+      5,
+      TEST_POINT_BYTE6_SMM_END_OF_DXE_SMRR_FUNCTIONAL
+      );
+  }
+
+  DEBUG ((DEBUG_INFO, "======== TestPointStandaloneMmEndOfDxe - Exit\n"));
+  return EFI_SUCCESS;
+}
+
+/**
+  This service verifies the security of Standalone MM communication buffers at Standalone MM Ready To Lock.
+
+  Test subject: Standalone MM communication buffer.
+  Test overview: Verify only CommBuffer and MMIO are mapped in the page table.
+  Reporting mechanism: Dumps the memory map and GCD map at StandaloneMmReadyToLock and checks at StandaloneMmReadyToBoot.
+
+  @retval EFI_SUCCESS         The test point check was performed successfully.
+  @retval EFI_UNSUPPORTED     The test point check is not supported on this platform.
+**/
+EFI_STATUS
+EFIAPI
+TestPointStandaloneMmReadyToLockSecureStandaloneMmCommunicationBuffer (
+  VOID
+  )
+{
+  if ((mFeatureImplemented[TEST_POINT_INDEX_BYTE6_SMM] & TEST_POINT_BYTE6_SMM_READY_TO_LOCK_SECURE_SMM_COMMUNICATION_BUFFER) == 0) {
+    return EFI_SUCCESS;
+  }
+
+  DEBUG ((DEBUG_INFO, "======== TestPointStandaloneMmReadyToLockSecureStandaloneMmCommunicationBuffer - Enter\n"));
+
+  //
+  // Collect information here, because it is last chance to access outside SMRAM.
```

Comment:
This is not really entirely true anymore :)

Also, should return unsupported?

Reply:
Fixed to return unsupported.